File: examples/datadog/expressjs-streaming/cdk/app/cdk.ts
import { App, CfnOutput, Duration, Stack, StackProps } from "aws-cdk-lib";
import {
  DockerImageFunction,
  DockerImageCode,
  FunctionUrlAuthType,
  Architecture,
  InvokeMode,
} from "aws-cdk-lib/aws-lambda";
import * as path from "path";

const app = new App();

class LwaStack extends Stack {
  constructor(scope: App, id: string, props?: StackProps) {
    super(scope, id, props);

    const lwa_lambda = new DockerImageFunction(this, id, {
      code: DockerImageCode.fromImageAsset(
        path.join(__dirname, "../../lambda-asset"),
        {
          cacheDisabled: true,
        }
      ),
      functionName: id + "-lambda",
      timeout: Duration.seconds(9),
      // Optional: Specify the architecture of the docker image if it's not X86_64
      architecture: Architecture.ARM_64,
    });

    const functionUrl = lwa_lambda.addFunctionUrl({
      authType: FunctionUrlAuthType.NONE,
      invokeMode: InvokeMode.RESPONSE_STREAM,
    });

    new CfnOutput(this, "LambdaFunctionUrl", {
      value: functionUrl.url,
      description: "The Lambda Function URL",
    });

    lwa_lambda.addEnvironment(
      "AWS_LWA_LAMBDA_RUNTIME_API_PROXY",
      "127.0.0.1:9002"
    );

    lwa_lambda.addEnvironment("AWS_LWA_INVOKE_MODE", "response_stream");
    lwa_lambda.addEnvironment("DD_TRACE_PARTIAL_FLUSH_MIN_SPANS", "1");
    lwa_lambda.addEnvironment("DD_TRACE_PARTIAL_FLUSH_ENABLED", "false");
    lwa_lambda.addEnvironment("DD_API_KEY", process.env.DD_API_KEY || "");
    lwa_lambda.addEnvironment("DD_SERVICE", id);
  }
}

new LwaStack(app, "lwa-stack", {});

File: examples/cdk/gradio/app/cdk.py
import os 
from constructs import Construct 
from aws_cdk import App, Stack,Duration,CfnOutput,Environment
from aws_cdk.aws_lambda import (
    DockerImageFunction,
    DockerImageCode,
    Architecture,
    FunctionUrlAuthType
)


class GradioLambdaFn(Stack):
    def __init__(self,scope:Construct,construct_id:str,**kwargs)->None:
        super().__init__(scope,construct_id,**kwargs)
        
        # Lambda Fn
        lambda_fn = DockerImageFunction(
                            self,
                            id="AWSLambdaAdapterGradioExample",
                            code=DockerImageCode.from_image_asset( directory= os.path.dirname(__file__) , file="Dockerfile"),
                            architecture=Architecture.X86_64,
                            timeout=Duration.minutes(10),
                            description="dockerfile_lambda_deploy_using_cdk",
        )

        # HTTP URL add
        fn_url = lambda_fn.add_function_url(auth_type=FunctionUrlAuthType.NONE)

        # print
        CfnOutput(self,id="functionURL",value=fn_url.url,description="cfn_output")


# My Environment
env = Environment(
    account= os.environ.get('CDK_DEFAULT_ACCOUNT'),
    region=os.environ.get('CDK_DEFAULT_REGION')
)

app = App()
gradio_lambda = GradioLambdaFn(app,"GradioLambda",env=env)
app.synth()